crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -567,7 +568,7 @@ impl<'a> Resolver<'a> {
         for (stmt_id, global_info) in self.interner.get_all_globals() {
             if global_info.local_id == self.path_resolver.local_module_id() {
                 let global_stmt = self.interner.let_statement(&stmt_id);
-                let definition = DefinitionKind::Global(global_stmt.expression);
+                let definition = DefinitionKind::Global(global_stmt.expression, None); // TODO
```

Comment:
what is the TODO for?

Reply:
The None is for the storage slot of these globals, and this information isn't available here. So ideally it should be re-added. I think we may need to fix the way we handle name resolution of globals so that they are already in scope like functions, rather than needing to be re-added/recreated manually in the scope of every function body.

Reply:
I've fixed this - i've updated the code to store storage slots on a GlobalInfo struct instead of in the definition kind. This lines up better with where we store additional data for other ids like Definitions, and avoids the TODO here.